sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -371,6 +371,8 @@ class ParquetFileFormat
 
     case udt: UserDefinedType[_] => supportDataType(udt.sqlType)
 
+    case _: CalendarIntervalType => true
```

Comment:
I suppose you could combine this with AtomicType above, but whatever

Reply:
Unfortunately, I cannot extend `AtomicType` because it supposes ordering.

Reply:
Oh I only meant you can use one case statement for both, i think, but it doesn't matter much.

Reply:
I see. I will combine them.